workshop/content/lab1.5_kafka.md

```diff
@@ -0,0 +1,274 @@
+
+```
+oc project    # Using project "user$USER_NUMBER" on server...
+```
+
+Let's also make sure your user number is set.
+
+```
+export USER_NUMBER=x    # replace x with your number
+```
+
+Now we can create our sink.
+
+```
+oc apply -f https://raw.githubusercontent.com/RedHatGov/serverless-workshop-code/main/kafka/kafka-sink.yml
+```
+
+Verify that it is there and wait until it is ready
+
+```
+oc get ksvc
+```
+
+```
+NAME            URL                                               LATESTCREATED      LATESTREADY        READY   REASON
+eventinghello   http://eventinghello-xyz.sandbox373.opentlc.com   eventinghello-v1   eventinghello-v1   True
+```
+
+When Serverless services are created they are initially spun up, which is why we can see the logs
+
+```
+stern eventinghello -c user-container
+```
+
+Note, if you didn't run the `stern` command within ~90s of creating the sink, the container might have spun back down, but that's ok, you can keep going.
+
+### Create Topic
+We have a Kafka topic, we should now go ahead and create the Kafka topic.
+
+```
+curl -sS https://raw.githubusercontent.com/RedHatGov/serverless-workshop-code/main/kafka/kafka-topic.yml | sed "s/USER_NUMBER/$USER_NUMBER/" | oc apply -f -
+```
+
+Verify it was created
+
+```
+oc get kafkatopics
+# a nice shortcut is `oc get kt`
+```
+
+```
+NAME                                                                        PARTITIONS   REPLICATION FACTOR
+my-topic-USER_NUMBER                                                        10           1
+```
+
+### Create KafkaSource Instance
+Now we will create a KafkaSource instance that ties our topic to our sink.
+
+```
+curl -sS https://raw.githubusercontent.com/RedHatGov/serverless-workshop-code/main/kafka/kafka-source-to-sink.yml | sed "s/USER_NUMBER/$USER_NUMBER/" | oc apply -f -
+```
+
+Verify that it was created.
+
+```
+oc get kafkasource
+```
+
+```
+NAME             TOPICS           BOOTSTRAPSERVERS                                      READY   REASON   AGE
+mykafka-source   ["my-topic-USER_NUMBER"]   ["my-cluster-kafka-bootstrap.kafka:9092"]   True             8s
+```
+
+Using the web browser, click on over to the `Developer` view.  Then click `Topology` and make sure you are in the correct `Project`.  You can then verify the KafkaSource is configured correctly.
+
+![Serverless Kafka Source](./images/serverless_kafka_source.png)
+
+### See It In Action
+Now let's see events flow through the Serverless system. Create a kafka producer.
+
+In one terminal run
+
+```
+oc run kafka-producer -ti --image=strimzi/kafka:0.19.0-kafka-2.5.0 --rm=true --restart=Never -- bin/kafka-console-producer.sh --broker-list my-cluster-kafka-bootstrap.kafka:9092 --topic my-topic-$USER_NUMBER
+```
+
+In another terminal run
+
+```
+stern eventinghello -c user-container
+```
+
+Every message you type into the producer, will appear in the knative service logs (that you are viewing through `stern`), along with other CloudEvent metadata.
+
```

Comment:
For example, if you enter `this is a test` in the terminal, you should see the stern log output: